### PR TITLE
[GPU] Bugfix 1c redun reorder

### PR DESCRIPTION
**Pull request by byungilm**
### Details:
 - Minor bugfix in layout optimizer for conv (bfyx->bfyx)
 - modify logic of removing reorder for fsv32 (only shallow input)

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 1]
```diff
@@ -1333,7 +1333,7 @@ format layout_optimizer::get_preferred_format(program_node& node) {
         auto& input = node.get_dependency(0);
         if (use_onednn_impls && input.get_output_layout().size.feature[0] == 1 && input.is_in_data_flow() && !input.is_constant()) {
             auto in_layout = input.get_output_layout();
-            if (in_layout.format == format::bfyx && output_layout.size.feature[0] >= 16)
+            if (in_layout.format == format::bfyx && output_layout.size.feature[0] >= 16 && expected != format::bfyx)
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:1336]: need to check whether the format is planar or not. `byxf` needs to be taken into consideration.

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 2]
```diff
@@ -697,6 +702,24 @@ bool layout_optimizer::deconvolution_b_fs_yx_fsv16_opt(layout const &input_layou
     return false;
 }
 
+bool layout_optimizer::needs_onednn_bfyx_to_blocked(format fmt_prev, format fmt_next, layout& prev_output_layout, const convolution_node& node) {
+    auto next_output_layout = node.get_output_layout();
+    if (prev_output_layout.data_type != next_output_layout.data_type)
+        return false;
+
+    // Target output_layour format
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:710]: nit: `layour`->`layout`

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp [hunk 3]
```diff
@@ -366,23 +366,25 @@ void remove_redundant_reorders::run(program& p) {
         p.remove_if_dangling(node);
     }
 
-    // Remove reorder for Convolution bfyx -> fs_b_yx_fsv32 (+ onednn: bfyx -> b_fs_yx_fsv32)
+    // Remove reorder for Convolution bfyx -> fs_b_yx_fsv32
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp:369]: could you update the comment properly?

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp (hunk 2 above)
Comment (byungilm) [reply inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp]: fixed

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp (hunk 3 above)
Comment (byungilm) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/remove_redundant_reorders.cpp]: Modified

File: inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp [hunk 4]
```diff
@@ -299,11 +298,14 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
...
+        // Remove Reorder to support mixed format convolutions of bsv32fsv16 or bsv32fsv32 output
+        if (next.is_type<convolution>() && (prev.is_type<eltwise>() || prev.is_type<quantize>() || prev.is_type<reshape>()) &&
             (fmt_prev == format::bfyx || fmt_prev == format::bs_fs_yx_bsv4_fsv2) &&
-            ((fmt_next == format::bs_fs_yx_bsv32_fsv32 && (prev_output_layout.size.feature[0] == 3 || prev_output_layout.size.feature[0] == 4)) ||
-            (fmt_next == format::bs_fs_yx_bsv32_fsv16 && (prev_output_layout.size.feature[0] == 3 || prev_output_layout.size.feature[0] == 4))))
-            return true;
+            (fmt_next == format::bs_fs_yx_bsv32_fsv16 || fmt_next == format::bs_fs_yx_bsv32_fsv32)) {
+            if ((prev_output_layout.size.feature[0] == 3 || prev_output_layout.size.feature[0] == 4) ||
+                (prev_output_layout.size.feature[0] % 32 == 0 && prev_output_layout.size.spatial[0] == 1 && prev_output_layout.size.spatial[1] == 1))
+                return true;
```
Comment (isanghao) [inline inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp:307]: This is for removing redundant reorder, right?
* Could you separate this logic from the conditional logic of first conv?
* What about putting into remove_redundant_reorders, instead?
* The next node type does not matter. It doesn't need to be conv

**Review by isanghao — APPROVED** [review]

**Pull request merged**